packages/editor/src/editor/sidebar/fields/BadgeField.tsx

```diff
@@ -0,0 +1,48 @@
+import { BasicField, InputBadge, InputBadgeArea, splitNewLine, type BasicFieldProps } from '@axonivy/ui-components';
+import { useEffect, useRef, useState } from 'react';
+import { badgeProps } from '../../../utils/badge-properties';
+
+export type BadgeFieldProps = BasicFieldProps & {
+  value: string;
+  childRef: React.RefObject<HTMLElement>;
+  area?: boolean;
+};
+
+export const BadgeField = ({ value, label, children, message, childRef, area, ...props }: BadgeFieldProps) => {
+  const [focus, setFocus] = useState(false);
+  const inputRef = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState(0);
+
+  const handleFocusLoss = (event: MouseEvent) => {
+    if (inputRef.current && !inputRef.current.contains(event.target as Node)) {
+      setFocus(false);
+    }
+  };
+
+  useEffect(() => {
+    setHeight(area ? splitNewLine(value).length * 14 : 15);
+    document.addEventListener('mousedown', handleFocusLoss);
+    return () => {
+      document.removeEventListener('mousedown', handleFocusLoss);
+    };
+  }, [value, area]);
+
+  const handleFocus = () => {
+    setFocus(true);
+    setTimeout(() => {
```

Comment:
I'm not quite sure why the setTimeout is necessary here. If possible, I’d consider exploring a solution that doesn't require it, as this can sometimes lead to unexpected bugs.

Reply:
If I try to instantly focus it after setting focus to true it can't be focused, because it's not been rendered.
I could solve it with a second useEffect hook..

```typescript
  useEffect(() => {
    if (focus && childRef.current) {
      childRef.current.focus();
    }
  }, [focus]);
```